SignalShareExtension/ShareViewController.swift

```diff
@@ -788,76 +875,108 @@ public class ShareViewController: UIViewController, ShareViewDelegate, SAEFailed
 
         itemProvider.loadItem(forTypeIdentifier: srcUtiType, options: nil, completionHandler: loadCompletion)
 
-        return promise.then { [weak self] (itemUrl: URL, utiType: String) -> Promise<SignalAttachment> in
-            guard let strongSelf = self else {
-                let error = ShareViewControllerError.obsoleteShare
-                return Promise(error: error)
-            }
-
-            let url: URL = try {
-                if strongSelf.isVideoNeedingRelocation(itemProvider: itemProvider, itemUrl: itemUrl) {
-                    return try SignalAttachment.copyToVideoTempDir(url: itemUrl)
-                } else {
-                    return itemUrl
-                }
-            }()
+        return promise
+    }
 
-            Logger.debug("building DataSource with url: \(url), utiType: \(utiType)")
+    private func buildAttachment(forLoadedItem loadedItem: LoadedItem) -> Promise<SignalAttachment> {
```

Comment:
Note that the huge `buildAttachment()` method has been decomposed into a series of methods which each correspond to a phase of the attachment loading process.